pkg/kubernetes/subsetdiff.go

```diff
@@ -32,59 +37,85 @@ func (k Kubectl) SubsetDiff(y string) (string, error) {
 			return "", errors.Wrap(err, "decoding yaml")
 		}
 
-		// filename
-		m := objx.New(util.CleanupInterfaceMap(rawShould))
-		name := strings.Replace(fmt.Sprintf("%s.%s.%s.%s",
-			m.Get("apiVersion").MustStr(),
-			m.Get("kind").MustStr(),
-			m.Get("metadata.namespace").MustStr(),
-			m.Get("metadata.name").MustStr(),
-		), "/", "-", -1)
-
-		// kubectl output -> current state
-		rawIs, err := k.Get(
-			m.Get("metadata.namespace").MustStr(),
-			m.Get("kind").MustStr(),
-			m.Get("metadata.name").MustStr(),
-		)
-		if err != nil {
-			if _, ok := err.(ErrorNotFound); ok {
-				rawIs = map[string]interface{}{}
-			} else {
-				return "", errors.Wrap(err, "getting state from cluster")
-			}
-		}
+		rs++
+		go subsetDiff(k, rawShould, results, errs)
 
-		should, err := yaml.Marshal(rawShould)
-		if err != nil {
-			return "", err
-		}
+	}
 
-		is, err := yaml.Marshal(subset(m, rawIs))
-		if err != nil {
-			return "", err
+	for rs > 0 {
+		select {
+		case d := <-results:
+			docs = append(docs, d)
+			rs--
```

Comment:
We should decrement it in errors also? More like `for i := 0; i < rs; i++` would be better.